lib/cardano-sl.cabal

```diff
@@ -45,10 +48,14 @@ library
 
                         -- Logic layer
                         Pos.Logic.Types
+                        Pos.Logic.Full
```

Comment:
This "Logic" is only used by the diffusion layer, so it's basically an interface for the diffusion layer. It can be moved to `infra` as well, and be implemented in `lib`. It's better for it to have `Diffusion.` prefixed.

Reply:
Absolutely not. Logic may be at the moment only used by the diffusion layer, but it has no intrinsic link to it; the two are completely different, unrelated, and should not be entangled.

Reply:
I do not really understand. If somebody wants to have some capabilities, he uses monadic constraints related to that part of the code. You decided that it is not the correct way to do things and it's totally fine (to use dictionaries)if it stays out of code covered by the general dev consensus.

Moreover, what you say doesn't contradict with my suggestion to move `Logic` to `infra`.

Reply:
> Moreover, what you say doesn't contradict with my suggestion to move Logic to infra.

Putting it in the `Diffusion` namespace makes no sense, I'm strongly opposed to that. Moving it to infra seems like a waste of time. What does it buy us? This is not something we should waste any time discussing.

Reply:
>Putting it in the Diffusion namespace makes no sense, I'm strongly opposed to that. 

1. There's no evidence that `Logic` would be used anywhere else at all. Every other code component does it great with monadic constraints.
2. The only single usecase of `Logic` now is as interface for the diffusion layer. 

>Moving it to infra seems like a waste of time. What does it buy us? This is not something we should waste any time discussing.

It's not the way people usually collaborate on big projects. Other people actually want code to be readable and properly structured, so every single PR change must be approved by everyone. Pushing PRs just before deadline w/o proper structure is a way to do, but definitely not the good one in the long term. Either you or me will be refactoring this code later anyway, so it's easier for you to do it now.

Also, i think it's super easy to keep modules in the packages they were located in previously.

Reply:
> Also, i think it's super easy to keep modules in the packages they were located in previously.

Sometimes it's not the best option.

> Other people actually want code to be readable and properly structured

Yes, this is my goal as well. It seems we just disagree on what that means. But important as they are, readability and proper structure truly are *less* important than a well-functioning product that users are happy with. My goal is to make this program better for the users; this pull request is an important step; we can't hold it up too long bike-shedding over where a module goes.

I find that most of the review comments I receive are nits about things like this. I'm happy to make the requested changes where they make sense and aren't too expensive, but please don't let these things hold up the review. I want us to move faster towards approval. Nits are fine to point out, but I need a sufficient condition for approval so I can make progress and know that there's a green light coming up soon.

> . Pushing PRs just before deadline w/o proper structure is a way to do, but definitely not the good one in the long term

I pushed this PR over a month ago and I'm still waiting for a sufficient condition for approval.